### PR TITLE
Add the Web basic security report

### PR DESCRIPTION
**Pull request by Donnype**
### Changes
Adds a web system basic security report.

### Issue link
Closes part of #2081

### Demo
image

---

### Code Checklist
- [x] All the commits in this PR are properly PGP-signed and verified;
- [x] This PR only contains functionality relevant to the issue; tickets have been created for newly discovered issues.
- [x] I have written unit tests for the changes or fixes I made.
- [x] For any non-trivial functionality, I have added integration and/or end-to-end tests.
- [x] I have performed a self-review of my code and refactored it to the best of my abilities.

### Communication
- [x] I have informed others of any required `.env` changes files if required and changed the `.env-dist` accordingly.
- [x] I have made corresponding changes to the documentation, if necessary.

---
## Checklist for code reviewers:
Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_code.md) into your comment.

---
## Checklist for QA:
Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_qa.md) into your comment.

### CODE REVIEW

**Comment by stephanie0x00** [comment]
## Checklist for QA:
- [x] I have checked out this branch, and successfully ran a fresh `make reset`.
- [x] I confirmed that there are no unintended functional regressions in this branch:
    - [x] I have managed to pass the onboarding flow
    - [x] Objects and Findings are created properly
    - [x] Tasks are created and completed properly
- [] I confirmed that the PR's advertised `feature` or `hotfix` works as intended.

### What works:
Onboarding flow works. 
It's possible to generate the basic security report and it shows the compliance checks for all hosts. 

### What doesn't work:
See below. 

### Bug or feature?:
- Not all hosts have the table with findings, my guess would be that no compliance data is present for these sites.
- The compliance issue table lines are shown twice, see the marking in the screenshot below. 
- Some of the compliance issues contain the text 'None' with the risk level 'Pending'. 
- It would be nice to have this list view sorted based on the risk level as that helps to filter out which require action first. 
- There might be a bug in the check or boefje for security.txt, as transip.nl seems to have a security.txt file, but the compliance check says 0/1. Maybe because it redirects to https://examplec.com/security.txt (without the .well-known folder). 

security-report

transip

transip-sec

**Comment by Donnype** [comment]
@stephanie0x00 I resolved most issues, but I am not sure what you mean by the first one? And I think the last one is due to a boefje issue since when I manually add a SecurityTXT object, the compliance issue is resolved for me..

**Comment by stephanie0x00** [comment]
So for some of the hosts in the report it doesnt show the compliance issue table below the table with compliance checks. So the table of the first screenshot isn't added to all hosts in the report. Does that make it more clear?

**Comment by Donnype** [comment]
@stephanie0x00 Ah right, indeed: most compliance check are the absence of a specific Finding, except the `SecurityTXT` and certificate checks where we look for `SecurityTXT` and `X509Certificate` OOIs. If the table is empty that means no Findings have been found, even when not all compliance has been fixed, as we have no recommendation and risk level mechanism for the `SecurityTXT` and `X509Certificate` issues. I added a check that only adds the table if there actually are elements to be shown.

Looking into this now I see that we can actually just leverage the `"KAT-NO-CERTIFICATE"` finding type, which should fix the latter one. This is fixed now, although we now have the same check for "offers https"... Perhaps we should consider a `"KAT-NO-SECURITY-TXT"` as well @noamblitz?

File: rocky/reports/report_types/web_system_report/report.py [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+    name = _("Web System Report")
+    description = _("Web system reports check web systems on basic security standards.")
+    plugins = {
+        "required": [
+            "nmap",
+            "dns-records",
+            "testssl-sh-ciphers",
+            "ssl-version",
+            "ssl-certificates",
+            "webpage-analysis",
```
Comment (underdarknl) [inline rocky/reports/report_types/web_system_report/report.py:85]: Shouldn' this also list the security.txt boefje?

**Review by underdarknl — APPROVED** [review]

**Pull request merged**